frontend/src/components/course-search/CourseRow.tsx

```diff
@@ -36,10 +37,51 @@ export default memo(function CourseRow(props: {
         props.updateDetailsSize(detailsBounds.height);
     }, [detailsBounds?.height, props.updateDetailsSize, props.expand]);
 
+    if (props.fromOtherTerm) {
```

Comment:
isn't this whole block essentially the same as the thing beneath it? why the code duplication?

Reply:
My idea was to separate the case where the `CourseRow` component represents the course from the non-current term out and display a much simpler version of `CourseRow` since it would not make sense for the `CourseRow` component to be expandable if it doesn't belong to this term (also, would affect course description map query to be much harder to handle). To achieve that, I wanted it serve as a anchor tag to their respective term for simplicity, so I use this if-check to render a more simple `CourseRow` element which is different from the existing `CourseRow`. 

I agree that there are quite a few duplications which could be cleaned up. I only worried that it would be overly complicated than having 2 cases separately like I did since the "foreign-term" `CourseRow` element are much simpler. (discussion needed)